### ISSUE TITLE
Support for task parameters

### ISSUE THREAD

**Issue by darsain** [robrich/orchestrator#17]
To reduce code duplication in tasks that do the same thing just a little bit differently, it'd be nice to be able to define some basic task parameters. Example implementation:

Task that accepts 1 optional parameter `bar` with a default value `baz`:

``` js
concert.add('foo:{bar=baz}', function (params) {
  console.log(params.bar);
});

concert.start('foo'); // logs "baz"
concert.start('foo:fam'); // logs "fam"
```

_(Maybe without curly braces, they are not really necessary.)_

Also a way to pass parameters as object so there would be no need to serialize the data when running from within the code and not gulp's command line:

``` js
concert.start('foo', { bar: 'fam' });
```
### Use case

For example the [gulp-bump](https://github.com/stevelacy/gulp-bump) task bumps the version property in json files. The bump type can be `patch`, `minor`, `major`, or `prerelease`.

Currently, if I want to be able to specify the bump type from command line, I need to define 4 different tasks that do the same thing. With proposition above I could just do:

``` js
gulp.task('bump:{type=patch}', function (params) {
  gulp.src('package.json')
    .pipe(bump({ type: params.type }))
    .pipe(gulp.dest('./'));
});
```

And call it like:

``` bash
gulp bump:minor
```

**Comment by lkraav** [comment]
@heikki thanks dude, you helped me figure out exactly what I needed: only re-compiling a specific less file in a varying directories, triggered by watching any less file. paljon olutta sinulle ja ystävillesi!

**Comment by heikki** [comment]
@lkraav :beers: 
Just beware that gulp.watch callback has this issue (gulp 3 & multiple watches): https://github.com/gulpjs/gulp/issues/776

**Comment by heikki** [comment]
Nvm, it doesn't seem to affect if you use only callback.

**Comment by jokeyrhyme** [comment]
My use case for this is that I have my tests split into separate files, and sometimes I only want to run specific tests. Seems like overkill to define separate tasks for each test.

**Comment by robrich** [comment]
Create a variable in your gulpfile and within the task use the variable -- similar to how you'd use a sources array to pass into gulp.src

**Comment by savager** [comment]
Apologies if I'm way off here, I'm new, but could you just change line 272:

`runTask(task.fn.bind(this), function (err, meta) {`

to 

`runTask(task.fn.bind(this, task.arguments), function (err, meta) {`

then I added a method to gulp: 

`gulp.constructor.prototype.addArguments= (task, obj) => {
	gulp.tasks[task].arguments = obj;
} 
gulp.addArguments('dev:css:desktop', {'platform':'desktop'});`

basically I'm doing this so I can dynamically generate gulp tasks based on a flattened object. I totally realize that a normal function would be fine and dandy.

**Comment by robrich** [comment]
That's an interesting idea.  You could also accomplish it like this:

	function dev_css_desktop(options) {
		// ...
	}

	gulp.task('dev:css:desktop', dev_css_desktop.bind(null, {'platform':'desktop'}));

**Comment by savager** [comment]
Ha! Thanks
